lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1205,6 +1214,44 @@ FIRRTLModuleLowering::lowerModule(FModuleOp oldModule, Block *topLevelModule,
   return newModule;
 }
 
+sv::InterfaceOp
+FIRRTLModuleLowering::lowerIntreface(FInterfaceOp interface,
+                                     Block *topLevelModule,
+                                     CircuitLoweringState &loweringState) {
+
+  // Map the ports over, lowering their types as we go.
+  SmallVector<PortInfo> firrtlPorts = interface.getPorts();
+  SmallVector<hw::PortInfo, 8> ports;
+  if (failed(lowerPorts(firrtlPorts, ports, interface, interface.getName(),
+                        loweringState)))
+    return {};
+
+  auto builder = OpBuilder::atBlockEnd(topLevelModule);
+  auto nameAttr = builder.getStringAttr(interface.getName());
+  auto newInterface =
+      builder.create<sv::InterfaceOp>(interface.getLoc(), nameAttr);
+  if (auto outputFile = interface->getAttr("output_file"))
+    newInterface->setAttr("output_file", outputFile);
+  if (auto comment = interface->getAttrOfType<StringAttr>("comment"))
+    newInterface.setCommentAttr(comment);
+  builder.setInsertionPointToEnd(newInterface.getBodyBlock());
+  for (auto &op : interface) {
+    if (auto wire = dyn_cast<WireOp>(op)) {
+      auto type = lowerType(wire.getType().cast<FIRRTLBaseType>(),
+                            topLevelModule, nameAttr.getValue());
+      builder.create<sv::InterfaceSignalOp>(interface.getLoc(),
+                                            wire.getNameAttr(), type);
+    } else if (auto verbatim = dyn_cast<sv::VerbatimOp>(op))
+      builder.create<sv::VerbatimOp>(
+          verbatim.getLoc(), verbatim.getFormatStringAttr(),
+          verbatim.getSubstitutions(), verbatim.getSymbolsAttr());
+    else {
+      op.emitOpError("Unhandled op inside firrtl.interface");
```

Comment:
Should this be checked in a verifier / is this true in general or just what's implemented here (are only wires and verbatim's supported inside these)?